packages/mui-base/src/useNumberField/useNumberField.types.ts

```diff
@@ -1,5 +1,90 @@
 import type { ScrubHandle } from './useScrub.types';
 
+export interface UseNumberFieldParameters {
+  /**
+   * The id of the input element.
+   */
+  id?: string;
+  /**
+   * The minimum value of the input element.
+   */
+  min?: number;
+  /**
+   * The maximum value of the input element.
+   */
+  max?: number;
+  /**
+   * The small step value of the input element when incrementing while the meta key is held. Snaps
+   * to multiples of this value.
+   * @default 0.1
+   */
+  smallStep?: number;
+  /**
+   * The step value of the input element when incrementing, decrementing, or scrubbing. It will snap
+   * to multiples of this value. When unspecified, decimal values are allowed, but the stepper
+   * buttons will increment or decrement by `1`.
+   */
+  step?: number;
+  /**
+   * The large step value of the input element when incrementing while the shift key is held. Snaps
+   * to multiples of this value.
+   * @default 10
+   */
+  largeStep?: number;
+  /**
+   * If `true`, the input element is required.
+   * @default false
+   */
+  required?: boolean;
+  /**
+   * If `true`, the input element is disabled.
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the input element is invalid.
+   * @default false
+   */
+  invalid?: boolean;
+  /**
+   * If `true`, the input element is focused on mount.
+   * @default false
+   */
+  autoFocus?: boolean;
+  /**
+   * If `true`, the input element is read only.
+   * @default false
+   */
+  readOnly?: boolean;
+  /**
+   * The name of the input element.
+   */
+  name?: string;
+  /**
+   * The raw number value of the input element.
+   */
+  value?: number | null;
+  /**
+   * The default value of the input element. Use when the component is not controlled.
+   */
+  defaultValue?: number;
+  /**
+   * Whether to allow the user to scrub the input value with the mouse wheel while focused and
+   * hovering over the input.
+   * @default false
+   */
+  allowWheelScrub?: boolean;
+  /**
+   * Options to format the input value.
+   */
+  format?: Intl.NumberFormatOptions;
+  /**
+   * Callback fired when the number value changes.
+   * @param {number | null} value The new value.
+   */
+  onChange?: (value: number | null) => void;
```

Comment:
To be consistent with native event handlers, we always place the event object (of the native event that triggered the change) as the first argument (and the value is usually the second one). This is what form libraries often expect.

Reply:
I dislike with that API because it's not ergonomic, and furthermore it's not possible because sometimes the change isn't triggered by an event because it auto-increments when holding a stepper button.  The current API aligns with react-aria

Reply:
Let's discuss this separately, as I feel we need to be consistent with the API across components (one way or another)